swan-lake/resources/featured-scenarios/write-a-restful-api-with-ballerina.md

```diff
@@ -28,76 +26,161 @@ This guide will walk you through creating a RESTful API with two endpoints. In t
+    - The `main.bal` file provides the look and feel of a simple Ballerina program. 
+
+2. Open the created package in VS Code.
 
-- `Ballerina.toml` is the file that makes the folder a Ballerina package. It also contains a test directory to include tests for the service. However, this will not be used in this guide. 
-- The `service.bal` template file provides a look and feel about Ballerina services. 
+3. Remove the generated content in the `main.bal` file and open the diagram view in VS Code.
+
+    ![Open diagram view](/learn/images/featured-scenarios/write-a-restful-api-with-ballerina/open-diagram-view.gif)
 
 ## Create the dataset
 
-To keep things simple, an in-memory dataset is used with three entries. To add the definition of the record and the declaration of the table, replace the API template file (i.e., `service.bal`) with the code below.
+An in-memory dataset with three entries is used to keep things simple. Follow the steps below to add the definition of the record and the declaration of the [table](/learn/by-example/table/) that holds the data.
+
+1. Generate the record types corresponding to the payload of the REST service by providing the record name as `CovidEntry` and the sample JSON object below.
+
+    >**Tip:** You need to complete the generated record by adding the pipe signs to mark the record as a closed one and adding the `readonly` descriptor to the `iso_code` field to make it non-modifiable as it is the key of the table, which cannot be represented in the JSON format.
+
+    ```json
+    {
+        "iso_code": "AFG",
+        "country": "Afghanistan",
+        "cases": 159303,
+        "deaths": 7386,
+        "recovered": 146084,
+        "active": 5833
+    }
+    ```
+    ![Create data record](/learn/images/featured-scenarios/write-a-restful-api-with-ballerina/create-data-record.gif)
+
+2. Create the table, as shown below.
+
+    >**Tip:** Enter `CovidEntry` as the table type, `iso_code` as the key, and `covidTable` as the variable type when creating the table.
+
+    ![Create data table](/learn/images/featured-scenarios/write-a-restful-api-with-ballerina/create-data-table.gif)
+
+3. Replace the `{key: value}` placeholder of the generated table with the following entries to add the initial data to the table.
+
+    ```
+        {
+            iso_code: "AFG", 
+            country: "Afghanistan", 
+            cases: 159303, 
+            deaths: 7386, 
+            recovered: 146084, 
+            active: 5833
+        },
+        {
+            iso_code: "SL", 
+            country: "Sri Lanka", 
+            cases: 598536, 
+            deaths: 15243, 
+            recovered: 568637, 
+            active: 14656
+        },
+        {
+            iso_code: "US", 
+            country: "USA", 
+            cases: 69808350, 
+            deaths: 880976, 
+            recovered: 43892277, 
+            active: 25035097
+        }
+    ```
+The generated record and the table will be as follows.
```

Comment:
Instead of expanded version, can we have single line initialization which is compact.

``` 
{iso_code: "AFG", country: "Afghanistan", cases: 159303, deaths: 7386, recovered: 146084, active: 5833},
{iso_code: "SL", country: "Sri Lanka", cases: 598536, deaths: 15243, recovered: 568637, active: 14656},
{iso_code: "US", country: "USA", cases: 69808350, deaths: 880976, recovered: 43892277, active: 25035097}
```

Reply:
Updated.